python/social_bot/gazebo_agent.py

```diff
@@ -188,27 +180,15 @@ def get_observation(self, teacher, sentence_raw="hello"):
             obs = self._create_observation_dict(teacher, sentence_raw)
         elif self._use_image_observation:  # observation is pure image
             obs = self.get_camera_observation()
+        elif self._goal_conditioned:
+            obs = self._generate_goal_conditioned_obs(teacher)
         else:  # observation is pure low-dimentional states
             obs = teacher.get_task_specific_observation(self)
-            if self._goal_conditioned:
-                flat_obs = obs
-                obs = OrderedDict()
-                obs['observation'] = flat_obs
-                obs['achieved_goal'] = flat_obs[6:8]
-                obs['desired_goal'] = flat_obs[12:14]
-                if self._use_aux_achieved:
-                    obs['observation'] = flat_obs[14:]
-                    obs['aux_achieved'] = np.concatenate(
-                        (flat_obs[:6], flat_obs[8:12]), axis=0)
-                    if self._xy_only_aux:
-                        # 2: z-speed, 3, 4: angular velocity, 5: yaw-vel, 6, 7: x, y, 8: z, 9, 10, 11: pose
-                        obs['observation'] = np.concatenate(
-                            (flat_obs[2:5], flat_obs[8:11], flat_obs[14:]),
-                            axis=0)
-                        obs['aux_achieved'] = np.concatenate(
-                            (flat_obs[0:2], np.expand_dims(flat_obs[5], 0),
-                             np.expand_dims(flat_obs[11], 0)),
-                            axis=0)
+        return obs
+
+    def _generate_goal_conditioned_obs(self, teacher):
+        tasks = teacher.get_task_groups()[0].get_tasks()
```

Comment:
do we need to assert here?

Reply:
I moved the assert in get_observation_space, which is only called once per run, and is called before getting obs.